### DISCUSSION TITLE
Not having _geo in displayedAttributes to return _geoDistance

### DISCUSSION THREAD

**Discussion by mateosuarezdev** [meilisearch/product#776] [Feedback & Feature Proposal]
Hi, there's another similar post from about a year ago but I wasn't able to find any solution and I think it wasn't addressed.
I really want to use meilisearch as the search engine for my app, however I need some places locations to stay private but still showing the distance between the users and those. If any user enters the client side network requests will be able to see the lat and lng geo points, is there any workaround to have the _geoDistance value returned for each document but not the _geo fields?

**Comment by macraig** [comment]
Hi @suarezlifestyle ,

Thank you for sharing your use case. Unfortunately, there isn't a current workaround that hides the `_geo` fields while still returning the `_geoDistance` value.

However, we're considering this scenario for future updates. One possible solution we're exploring is introducing a rule-based syntax for `displayedAttributes`. With this new syntax, you could control whether `_geoDistance` is displayed without exposing `_geo`.

Here’s a rough proposal of how it might look:

```displayedAttributes: [
  {
    "display": "always",
    "attributePatterns": ["_geoDistance"]
  },
  {
    "display": "never",
    "attributePatterns": ["_geo"]
  },
  {
    "display": "always",
    "attributePatterns": ["*"]
  }
]
```

Would this approach fit your needs? Let us know your thoughts!

**Reply by mateosuarezdev** [reply]
Yes, if you can control whether or not expose geo while still getting geoDistance back it's indeed a good approach
